docs/architecture/0e-plans.md

```diff
@@ -0,0 +1,474 @@
+
+Due to a lack of bandwidth, we are not planning any major changes to most platform interfaces. As of the 0ε target, the following platform-related headers remain public.
+
+* `memory_buffer_alloc.h`: a feature used by some applications, with close to 0 maintenance cost for us.
+* `platform.h`, `platform_time.h`, `platform_util.h`: platform abstractions. We would like to adapt them, but we do not think we will have time in before 1.0, so this will have to wait until the next major version.
+* `threading.h`: We want to change the threading abstraction, but this is out of scope of the 0ε project. https://github.com/Mbed-TLS/mbedtls/issues/8455
+
+### Plans for hiding
+
+In an exposed header, the minimum work for 0ε is:
+
+* Ensure that exposed interfaces are not listed in the rendered Doxygen documentation.
+* Ensure that applications cannot inadvertently call private functions that are declared in exposed headers.
+
+#### TF-PSA-Crypto header locations
+
+We distinguish between three categories of headers:
+
+* Public headers define public, stable APIs.
+* Exposed headers define exposed interfaces, as well as private interfaces used by Mbed TLS. They need to be present when building Mbed TLS.
+* Private headers are only needed when building TF-PSA-Crypto itself.
+
+The following table summarizes the characteristics of each category.
+
+| Category | Location | In include path? |
+| -------- | -------- | ---------------- |
+| Public   | `include` | yes |
+| Exposed  | `include` | yes except for Doxygen |
+| Private  | `drivers/*/include` | only for crypto |
+
+TODO: Task to set up the Doxygen build
+
+#### Hiding functions in an exposed header
+
+For TF-PSA-Crypto 0ε, all private functions defined in private headers will be guarded by `defined(MBEDTLS_DECLARE_PRIVATE_IDENTIFIERS)`. We define this macro when building TF-PSA-Crypto and Mbed TLS, but applications should not define it.
+
+This is similar to `MBEDTLS_ALLOW_PRIVATE_ACCESS` to “bless” access to structure fields.
+
+Prototype: https://github.com/Mbed-TLS/TF-PSA-Crypto/pull/132
+
+Thus, for the 0ε milestone, in each affected header, we just need to place `#ifdef` guards around the function declarations.
+
+Rationale:
+
+* This is a small amount of work.
+* This requires few changes to the code, and they are very localized, so it will not disrupt other work happening in parallel.
+
+TODO: this leaves Doxygen comments around
+
+#### Separating private interfaces from exposed interfaces
+
+If a private interface of TF-PSA-Crypto is declared in an exposed header, in the medium term, we should move it to a private header. Note that we can only do that if the interface is not called from Mbed TLS code.
+
+When none the functions declared in an exposed header file are either public or called by Mbed TLS, we can move all the declarations gated by `MBEDTLS_DECLARE_PRIVATE_IDENTIFIERS` to a private header file and adjust `#include` directives accordingly. This can be done by a script.
```

Comment:
Wording: I found "when none [...] are either [...] or [...]" hard to parse. (I you're curious why, I think I interpreted "either A or B" as "A or B but not both" then putting "none" in front resulted in a double negation and that was too much for me. On 2nd though, I think that's not that either means here, but rather "when all of the functions [...] are neither public nor used by Mbed TLS"?)

Still, I'm not sure I get the first part of the sentence: since the rest of the sentence is about moving functions gated by `MBEDTLS_DECLARE_PRIVATE_IDENTIFIERS`, which by definition are not public, why mention public in the first place?

All this to say, I'm a bit confused, can you clarify?